### PR TITLE
Node testing mock overhaul

### PR DESCRIPTION
**Pull request by bmahall**
This PR makes the following changes to msal-node testing:--

TestFixtures.ts : Common mocks for usage , ex: Authority Factory, Telemetry Manager, etc.

For Client test classes:-
-Removes jest.mock("@azure/msal-common")
-Uses mock Authority & AuthorityFactory-createDiscoveredInstance's mock from 'test-fixtures.ts'
-Spies on AuthorizationCodeClient from common
-Mock implementation for various TokenClients

For Cache test classes:-
-Added more tests based on the Coverage report.

Storage.spec.ts:-
Tests supporting->
i) Change event being emitted when ChangeEmitter is registered
ii) Coverage for Catch blocks for setter and getter for Account, AccessToken, IdToken, RefreshToken

TokenCache.spec.ts:-
Tests supporting->
i) Edge cases like deserializing an empty cache, empty KVStore in case of empty Cache.

### CODE REVIEW

File: change/@azure-msal-node-03eb944d-e367-43d4-8dc3-931b2e5aaf90.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Node testing overhaul",
```
Comment (jasonnutter) [inline change/@azure-msal-node-03eb944d-e367-43d4-8dc3-931b2e5aaf90.json]:
```suggestion
  "comment": "Node testing overhaul #4433",
```

File: change/@azure-msal-node-03eb944d-e367-43d4-8dc3-931b2e5aaf90.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (jasonnutter) [inline change/@azure-msal-node-03eb944d-e367-43d4-8dc3-931b2e5aaf90.json]:
```suggestion
  "type": "none",
```

Comment (jasonnutter) [reply change/@azure-msal-node-03eb944d-e367-43d4-8dc3-931b2e5aaf90.json]: No functional changes to the library, only tests.

File: change/@azure-msal-node-03eb944d-e367-43d4-8dc3-931b2e5aaf90.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "Node testing overhaul",
+  "packageName": "@azure/msal-node",
+  "email": "bmahal@microsoft.com",
+  "dependentChangeType": "patch"
```
Comment (jasonnutter) [inline change/@azure-msal-node-03eb944d-e367-43d4-8dc3-931b2e5aaf90.json]:
```suggestion
  "dependentChangeType": "none"
```

**Review by jasonnutter — APPROVED** [review]
A couple proposals for the change file, otherwise looks good!

**Review by samuelkubai — APPROVED** [review]
LGTM
